docs/advanced/debugging.rst

```diff
@@ -27,15 +27,335 @@ And, of course, if you're hitting that ever-challenging ``No scope with a Tag ma
+                    -> λ:System.String
+                    X- λ:System.String
+                  X- DisposalTrackingMiddleware
+                X- ActivatorErrorHandlingMiddleware
+              X- RegistrationPipelineInvokeMiddleware
+            X- SharingMiddleware
+          X- ScopeSelectionMiddleware
+        X- CircularDependencyDetectorMiddleware
+      }
+      Resolve Request FAILED
+        System.DivideByZeroException: Attempted to divide by zero.
+          at MyProject.MyNamespace.MyMethod.<>c.<GenerateSimpleTrace>b__6_0(IComponentContext x) in /path/to/MyCode.cs:line 39
+          at Autofac.RegistrationExtensions.<>c__DisplayClass39_0`1.<Register>b__0(IComponentContext c, IEnumerable`1 p)
+          at Autofac.Builder.RegistrationBuilder.<>c__DisplayClass0_0`1.<ForDelegate>b__0(IComponentContext c, IEnumerable`1 p)
+          at Autofac.Core.Activators.Delegate.DelegateActivator.ActivateInstance(IComponentContext context, IEnumerable`1 parameters)
+          at Autofac.Core.Activators.Delegate.DelegateActivator.<ConfigurePipeline>b__2_0(ResolveRequestContext ctxt, Action`1 next)
+          at Autofac.Core.Resolving.Middleware.DelegateMiddleware.Execute(ResolveRequestContext context, Action`1 next)
+          at Autofac.Core.Resolving.Pipeline.ResolvePipelineBuilder.<>c__DisplayClass14_0.<BuildPipeline>b__1(ResolveRequestContext ctxt)
+          at Autofac.Core.Resolving.Middleware.DisposalTrackingMiddleware.Execute(ResolveRequestContext context, Action`1 next)
+          at Autofac.Core.Resolving.Pipeline.ResolvePipelineBuilder.<>c__DisplayClass14_0.<BuildPipeline>b__1(ResolveRequestContext ctxt)
+          at Autofac.Core.Resolving.Middleware.ActivatorErrorHandlingMiddleware.Execute(ResolveRequestContext context, Action`1 next)
+    }
+    Operation FAILED
+
+See how the return trip through the middleware changed to a ``X-`` in there? We know the error happened during executing the lambda. You can use these hints to see exactly where in the pipeline things are failing.
+
+DOT Graph Tracer
+================
+
+In addition to the ``DefaultDiagnosticTracer`` we provide a graphing tracer in the ``Autofac.Diagnostics.DotGraph`` package.
+
+If you add a reference to this package, you'll have the ability to trace out full dependency trees in a visual fashion using `DOT language <https://graphviz.org/doc/info/lang.html>`_. You can then render images with a tool like `Graphviz <https://graphviz.org/>`_.
+
+First, just like with the ``DefaultDiagnosticTracer``, register it with your container. This time the trace output will be a DOT graph.
+
+.. sourcecode:: csharp
+
+    // Build your container as you normally would.
+    var containerBuilder = new ContainerBuilder();
+    containerBuilder.RegisterType<Component>().As<IService>();
+    var container = containerBuilder.Build();
+
+    // Create a DOT graph tracer instance. The
+    // trace content will be DOT graph format.
+    var tracer = new DefaultDiagnosticTracer();
```

Comment:
Still referencing the Default tracer here, pretty sure it should be the DOT one.